modules/wallet/database.go

```diff
@@ -204,6 +208,47 @@ func dbDeleteSpentOutput(tx *bolt.Tx, id types.OutputID) error {
 	return dbDelete(tx.Bucket(bucketSpentOutputs), id)
 }
 
+// bucketAddrTransactions is special: it's okay if the key doesn't exist, and
+// we need to avoid inserting duplicate slice elements.
+
+func dbPutAddrTransactions(tx *bolt.Tx, addr types.UnlockHash, txns []uint64) error {
+	return dbPut(tx.Bucket(bucketAddrTransactions), addr, txns)
+}
+func dbGetAddrTransactions(tx *bolt.Tx, addr types.UnlockHash) (txns []uint64, err error) {
+	err = dbGet(tx.Bucket(bucketAddrTransactions), addr, &txns)
+	if err == errNoKey {
+		err = nil // no key means no transactions associated yet
+	}
+	return
+}
+func dbAddAddrTransaction(tx *bolt.Tx, addr types.UnlockHash, txn uint64) error {
+	txns, err := dbGetAddrTransactions(tx, addr)
+	if err != nil {
+		return err
+	}
+	for _, i := range txns {
```

Comment:
This could get bad if we end up with a lot of transactions coming to the same address. Not sure what bittrex's load looks like, and at least it's better than doing a linear operation using a disk iterator, but we could put a binary search here too.

We want to handle lopsided situations just as well as normal ones.

Reply:
It's linear, but decoding the transactions is also linear, and its constant factor is waaaay larger. If we need to optimize this call, the first thing to target is the decoding.